sw/device/lib/dif/dif_aon_timer.h

```diff
@@ -0,0 +1,274 @@
+   */
+  kDifAonTimerOk = kDifAonTimerOk,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifAonTimerError = kDifAonTimerError,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occurred.
+   */
+  kDifAonTimerBadArg = kDifAonTimerBadArg,
+  /**
+   * Indicates that this operation has been locked out, and can never
+   * succeed until hardware reset.
+   */
+  kDifAonTimerLocked = 3,
+} dif_aon_timer_watchdog_result_t;
+
+/**
+ * An Always-On Timer interrupt request type.
+ */
+typedef enum dif_aon_timer_irq {
+  /**
+   * Wake-up timer has expired.
+   */
+  kDifAonTimerWakeupThreshold = 0,
+  /**
+   * Watchdog timer "Bark" threshold has expired.
+   */
+  kDifAonTimerWatchdogBarkThreshold,
+} dif_aon_timer_irq_t;
+
+/**
+ * Creates a new handle for Always-On Timer.
+ *
+ * This function does not actuate the hardware.
+ *
+ * @param params Hardware instantiation parameters.
+ * @param[out] handle Out param for the initialised handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_init(dif_aon_timer_params_t params,
+                                          dif_aon_timer_t *handle);
+
+/** Starts Always-On Timer (wake-up timer).
+ *
+ * @param handle A Always-On Timer handle.
+ * @param threshold Threshold in ticks.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_wakeup_start(const dif_aon_timer_t *handle,
+                                                  uint32_t threshold);
+
+/** Stops Always-On Timer (wake-up timer).
+ *
+ * Stops the timer. Configuration is not touched, and can be restarted via
+ * `dif_aon_timer_wakeup_restart`.
+ *
+ * @param handle A Always-On Timer handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_wakeup_stop(const dif_aon_timer_t *handle);
+
+/** Restarts Always-On Timer (wake-up timer).
+ *
+ * Clears the counter and restarts the timer using the existing configuration.
+ *
+ * @param handle A Always-On Timer handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_wakeup_restart(
+    const dif_aon_timer_t *handle);
+
+/** Retrieves Always-On Timer (wake-up timer) tick count.
+ *
+ * @param handle An Always-On Timer handle.
+ * @param[out] count Current timer tick count.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_wakeup_get_count(
+    const dif_aon_timer_t *handle, uint32_t *count);
+
+/** Starts Always-On Timer (watchdog timer).
+ *
+ * @param handle A Always-On Timer handle.
+ * @param bark_threshold "Bark" threshold in ticks.
+ * @param bite_threshold "Bite" threshold in ticks.
+ * @param pause_in_sleep Watchdog is paused when device is in one of low power
```

Comment:
`is in one of low power` -> `is in one of the low power`

Reply:
Done.